src/components/ProfileCard.tsx

```diff
@@ -0,0 +1,67 @@
+import {User} from "next-auth";
+import {api} from "~/utils/api";
+import {timeAgo} from "~/utils/time";
+import {ProfilePicture} from "./ProfilePicture";
+import {Spinner} from "./Spinner";
+
+interface ProfileCardProps {
+    dateJoined?: Date,
+    name: string
+    imgUrl?: string | null,
+    userId: string,
+}
+
+export const ProfileCard: React.FC<ProfileCardProps> = ({
+    dateJoined,
+    name,
+    imgUrl,
+    userId,
+}) => {
+    const { data: nBlogPosts } = api.blogPost.count.useQuery({
+        userId: userId 
+    });
+    const { data: nComments } = api.comment.count.useQuery({
+        userId: userId
+    });
+
+    let isLoading = false;
```

Comment:
Can we move this to a `useState`? Using local variables to keep track of state within a component almost always leads to weird behaviour due to how react re-renders when props change.

Reply:
Here's an example of a useState declaration: 

https://github.com/CPSC319-2022/Continuus/blob/7c1b0961b2e7dab1fcd320d24ac21b98c2838435/src/components/BlogPostViewer.tsx#L12

Here's an example of setting a variable using `setState`

https://github.com/CPSC319-2022/Continuus/blob/7c1b0961b2e7dab1fcd320d24ac21b98c2838435/src/components/BlogPostViewer.tsx#L45

Reply:
took altay's suggestion